core/templates/pages/exploration-editor-page/modal-templates/state-diff-modal.component.ts

```diff
@@ -43,11 +43,17 @@ interface mergeviewOptions {
 })
 export class StateDiffModalComponent
   extends ConfirmOrCancelModal implements OnInit {
-  newState: State | null;
-  oldState: State | null;
-  newStateName: string;
-  oldStateName: string;
-  headers: headersAndYamlStrs;
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+
+  // The following properties will be null when there is no change introduced
+  // in either of the state. It remains same as original State.
+  @Input() newState!: State | null;
```

Comment:
These properties are passed from version-diff-visualization.component, so we need to use @Input(). I don't know how it was working without that.